### ISSUE TITLE
dict not included in SummaryTracker.print_diff()

### ISSUE THREAD

**Issue by daveisfera** [pympler/pympler#97]
`dict` variables and the values inside of them are not output when calling `print_diff()` with a `SummaryTracker`. Here's an example that demonstrates the behavior:

```
from pympler import tracker
from datetime import datetime

tr = tracker.SummaryTracker()
print('start')
tr.print_diff()

print('empty list')
list_test = []
tr.print_diff()

print('add to list')
list_test.append(datetime.now())
tr.print_diff()

print('empty dict')
dict_test = {}
tr.print_diff()

print('add to dict')
dict_test['test'] = datetime.now()
tr.print_diff()
```

Tested with 2.7.15 and 3.7.2 and neither output the any information about `dict`

**Comment by Pankrat** [comment]
Thanks for the report and the script to reproduce the issue. Not what I would expect here as well.

Turns out that the dict in your example is not tracked by the garbage collector (which is used internally to generate those diffs). If you add `dict_test['test'] = []`, the dict will be reported properly. More information can be found here: https://docs.python.org/3/library/gc.html#gc.is_tracked

With the current approach, I don't think there's anything that can be done about this behaviour. Sorry for the late answer.

**Comment by daveisfera** [comment]
Is there any way to disable this optimization so an issue can be tracked/diagnosed?

**Comment by Pankrat** [comment]
Hm, actually the object should be found via the referring global container. Looks like there's a bug in `muppy._is_containerobject`. I'll re-open the ticket and will test a fix. Thanks for persevering.

**Comment by daveisfera** [comment]
Glad to hear that the source of the issue was found

**Comment by Pankrat** [comment]
Should be fixed in Pympler 0.8. Thanks again for the report!
